### ISSUE TITLE
black-nb does not preserve encoding

### ISSUE THREAD

**Issue by mbaudin47** [tomcatling/black-nb#47]
black-nb can sometimes destroy a IPYNB file when this file contains UTF8 accents.

I work on a Ipynb with UTF8 accents. This corresponds to the before.ipynb.txt Notebook:

[before.ipynb.txt](https://github.com/tomcatling/black-nb/files/4575186/before.ipynb.txt)

Then I execute black-nb on it:

```
(C:\Users\myname\AppData\Local\Continuum\anaconda3-2019-03) Z:\myworkingdirectory>black-nb Untitled.ipynb
    1 cell left unchanged.
All done!
1 file left unchanged.
```

No cell was apparently changed. However, the new notebook:

[after.ipynb.txt](https://github.com/tomcatling/black-nb/files/4575188/after.ipynb.txt)

cannot be opened:

![image](https://user-images.githubusercontent.com/31351465/80976570-e712f400-8e23-11ea-99f7-01df743ef189.png)

This is because the cell contains a "é" which requires UTF8 encoding. 

![image](https://user-images.githubusercontent.com/31351465/80976367-ab782a00-8e23-11ea-99d6-c5e3dfb7aba4.png)

But the new file has ANSI encoding.

![image](https://user-images.githubusercontent.com/31351465/80976292-8c799800-8e23-11ea-87ff-d75a39e6bf05.png)

**Comment by liamcoatman** [comment]
Thanks for the bug report @mbaudin47. I think the issue is with using the default encoding for windows when we write out, instead of `utf-8`. I've opened a PR #48 which I think might solve the issue. I'm not able to test this because I don't have access to a windows machine. It would be helpful if you could test out this branch. To do this:

1. `git clone https://github.com/tomcatling/black-nb.git`
2. `cd black-nb`
3. `git checkout fix-decoding-error`
4. `pip install .`

Then see if you still have the same issue. Thanks

**Comment by mbaudin47** [comment]
Thank you very much for your quick fix. 

I tested it on Windows: it now works perfectly:

![image](https://user-images.githubusercontent.com/31351465/81042463-75cd5280-8eb0-11ea-9389-0365863e8d02.png)

However, the fix seems tricky to me:
```
        with src.open("w") as fp:
            nbformat.write(src_contents, fp)
```
works, but writes in ANSI, while:
```
        nbformat.write(src_contents, str(src))
```
works and preserves encoding? This seems a fragile fix to me, because both work.

I suggest to add a unit test which checks the encoding. This will make the fix stay in the future. This can be done with:
```
file --mime-encoding filename
```
You can install it with:
```
pip install python-magic
```
I guess that, without the fix, the encoding is changed even on Linux.

By the way, if you need Windows CI, you may use appveyor. Here is an example for OpenTURNS:

https://ci.appveyor.com/project/openturns/openturns

**Comment by liamcoatman** [comment]
Fixed in #52

**Comment by liamcoatman** [comment]
The reason we switched to passing a string to `nbformat.write` is that this function explicitly sets `encoding='utf-8'`.
